### PR TITLE
#8: Logging configuration:

### PR DESCRIPTION
**Pull request by PavelKostyrko**
- it was created project for logging
- it was created and configured model
- it was created DbContext and Configured DB connection
- it was created event-class and dictionary for it
- it was configured main Progam-file for new changes
- it was configured localization for logging

### CODE REVIEW

File: FitnessApp.Logic/Services/NutrientCategoryService.cs [hunk 1]
```diff
@@ -99,6 +105,7 @@ public async Task CreateAsync(NutrientCategoryDto nutrientCategoryDto)
             }
             catch
```
Comment (Eugene-Patysh) [inline FitnessApp.Logic/Services/NutrientCategoryService.cs]: in all places
catch(Exception ex) {
        _eventBus.Publish(new LogEvent(Statuses.Fail, "Creation", nutrientCategoryDto.GetType().Name.Replace("Dto", ""), $"Changes was not saved in data base: {ex.Message}"));
}

File: FitnessApp.Logic/Services/ProductSubCategoryService.cs [hunk 2]
```diff
@@ -94,6 +100,7 @@ public async Task CreateAsync(ProductSubCategoryDto productSubCategoryDto)
 
             try
             {
+                _eventBus.Publish(new LogEvent(Statuses.Fail, "Creation", productSubCategoryDto.GetType().Name.Replace("Dto", ""), "Changes was not saved in data base"));
```
Comment (Eugene-Patysh) [inline FitnessApp.Logic/Services/ProductSubCategoryService.cs]: ???

File: FitnessApp.Logic/Services/TreatingTypeService.cs [hunk 3]
```diff
@@ -94,6 +100,7 @@ public async Task CreateAsync(TreatingTypeDto treatingTypeDto)
 
             try
             {
+                _eventBus.Publish(new LogEvent(Statuses.Fail, "Creation", treatingTypeDto.GetType().Name.Replace("Dto", ""), "Changes was not saved in data base"));
```
Comment (Eugene-Patysh) [inline FitnessApp.Logic/Services/TreatingTypeService.cs]: ???

File: FitnessApp.Logging/Events/LogEvent.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+namespace FitnessApp.Logging.Events
+{
+    public class LogEvent : IntegrationEvent
+    {
+        public Statuses Status { get; set; }
+        public string Action { get; set; }
+        public string EntityType { get; set; }
+        public string Body { get; set; }
+
+        public LogEvent(Statuses status, string action, string entityType, string body)
```
Comment (Eugene-Patysh) [inline FitnessApp.Logging/Events/LogEvent.cs]: move JsonSerializer.Serialize to LogEvent

Body = JsonSerializer.Serialize(body);

File: FitnessApp.Web/Controllers/NutrientCategoryController.cs [hunk 5]
```diff
@@ -114,6 +122,7 @@ public async Task UpdateAsync([FromBody] NutrientCategoryDto nutrientCategoryDto
             _validator.Validate(nutrientCategoryDto, "UpdateNutrientCategory");
 
             await _nutrientCategoryService.UpdateAsync(nutrientCategoryDto);
+            _eventBus.Publish(new LogEvent(Statuses.Success, "Update", nutrientCategoryDto.GetType().Name.Replace("Dto", ""), JsonSerializer.Serialize(nutrientCategoryDto)));
```
Comment (Eugene-Patysh) [inline FitnessApp.Web/Controllers/NutrientCategoryController.cs]: move JsonSerializer.Serialize to LogEvent

Comment (Eugene-Patysh) [reply FitnessApp.Web/Controllers/NutrientCategoryController.cs]: for all

File: FitnessApp.Web/Controllers/NutrientController.cs [hunk 6]
```diff
@@ -114,6 +122,7 @@ public async Task UpdateAsync([FromBody] NutrientDto nutrientDto)
             _validator.Validate(nutrientDto, "UpdateNutrient");
 
             await _nutrientService.UpdateAsync(nutrientDto);
+            _eventBus.Publish(new LogEvent(Statuses.Success, "Update", nutrientDto.GetType().Name.Replace("Dto", ""), JsonSerializer.Serialize(nutrientDto)));
```
Comment (Eugene-Patysh) [inline FitnessApp.Web/Controllers/NutrientController.cs]: nutrientDto.GetType().Name.Replace("Dto", "") wtf?

add const ENTITY_TYPE

File: FitnessApp.Web/Controllers/ProductCategoryController.cs [hunk 7]
```diff
@@ -93,9 +100,17 @@ public async Task<ProductCategoryDto> GetByIdAsync(int? productCategoryId)
         [SwaggerRequestExample(typeof(ProductCategoryDto), typeof(ProductCategoryCreateExample))]
         public async Task CreateAsync ([FromBody] ProductCategoryDto productCategoryDto)
         {
+        // Example: Recording of validation fails
+            //if (!_validator.IsValid(productCategoryDto, "AddProductCategory", out string message))
+            //{
+            //    _eventBus.Publish(new LogEvent(Statuses.IsNotValid, "Creation", productCategoryDto.GetType().Name, JsonSerializer.Serialize(productCategoryDto)));
+            //    throw new Exception(message);
```
Comment (Eugene-Patysh) [inline FitnessApp.Web/Controllers/ProductCategoryController.cs]: ValidationException(message)

File: FitnessApp.Web/Controllers/ProductCategoryController.cs [hunk 8]
```diff
@@ -115,6 +130,7 @@ public async Task UpdateAsync([FromBody] ProductCategoryDto productCategoryDto)
             _validator.Validate(productCategoryDto, "UpdateProductCategory");
 
             await _productCategoryService.UpdateAsync(productCategoryDto);
+            _eventBus.Publish(new LogEvent(Statuses.Success, "Update", productCategoryDto.GetType().Name.Replace("Dto",""), JsonSerializer.Serialize(productCategoryDto)));
```
Comment (Eugene-Patysh) [inline FitnessApp.Web/Controllers/ProductCategoryController.cs]: productCategoryDto.GetType().Name.Replace("Dto","") could you use 100500 symbols and methods instead "productCategory"?

File: FitnessApp.Web/Controllers/ProductNutrientController.cs [hunk 9]
```diff
@@ -94,6 +101,7 @@ public async Task CreateAsync([FromBody] ProductNutrientDto productNutrientDto)
             _validator.Validate(productNutrientDto, "AddproductNutrient");
 
             await _productNutrientService.CreateAsync(productNutrientDto);
+            _eventBus.Publish(new LogEvent(Statuses.Success, "Creation", productNutrientDto.GetType().Name.Replace("Dto", ""), JsonSerializer.Serialize(productNutrientDto)));
```
Comment (Eugene-Patysh) [inline FitnessApp.Web/Controllers/ProductNutrientController.cs]: productNutrientDto.GetType().Name.Replace("Dto", "")
I can't say enough: it's brilliant!

File: FitnessApp.Logic/Services/ProductSubCategoryService.cs (hunk 2 above)
Comment (PavelKostyrko) [reply FitnessApp.Logic/Services/ProductSubCategoryService.cs]: Sry, my bad.

File: FitnessApp.Logic/Services/TreatingTypeService.cs (hunk 3 above)
Comment (PavelKostyrko) [reply FitnessApp.Logic/Services/TreatingTypeService.cs]: Sry, my bad.

File: FitnessApp.Web/Controllers/ProductCategoryController.cs (hunk 7 above)
Comment (PavelKostyrko) [reply FitnessApp.Web/Controllers/ProductCategoryController.cs]: Done.

File: FitnessApp.Logic/Services/NutrientCategoryService.cs (hunk 1 above)
Comment (PavelKostyrko) [reply FitnessApp.Logic/Services/NutrientCategoryService.cs]: Done

File: FitnessApp.Logging/Events/LogEvent.cs (hunk 4 above)
Comment (PavelKostyrko) [reply FitnessApp.Logging/Events/LogEvent.cs]: It is excellent decision! Done.

File: FitnessApp.Web/Controllers/NutrientCategoryController.cs (hunk 5 above)
Comment (PavelKostyrko) [reply FitnessApp.Web/Controllers/NutrientCategoryController.cs]: It is excellent decision! Done.

File: FitnessApp.Web/Controllers/ProductCategoryController.cs (hunk 8 above)
Comment (PavelKostyrko) [reply FitnessApp.Web/Controllers/ProductCategoryController.cs]: I can!

File: FitnessApp.Web/Controllers/ProductNutrientController.cs (hunk 9 above)
Comment (PavelKostyrko) [reply FitnessApp.Web/Controllers/ProductNutrientController.cs]: Thx u very much! It is very important to know, that my work was appreciated.

File: FitnessApp.Web/Controllers/NutrientController.cs (hunk 6 above)
Comment (PavelKostyrko) [reply FitnessApp.Web/Controllers/NutrientController.cs]: Done.

**Review by Eugene-Patysh — APPROVED** [review]

**Pull request merged**